File: app/src/main/java/fr/newglace/notedesnazes/Styles/Dialog/OptionNote.java
package fr.newglace.notedesnazes.Styles.Dialog;

import android.app.Activity;
import android.app.Dialog;
import android.util.DisplayMetrics;
import android.view.View;
import android.widget.ImageView;
import android.widget.Space;
import android.widget.TextView;
import fr.newglace.notedesnazes.R;
import fr.newglace.notedesnazes.Styles.reSize2;

public class OptionNote extends Dialog {
    private ImageView delete, password, favorite, favoriteColor, imageView5;
    private TextView lockText;

    public OptionNote(Activity activity) {
        super(activity, R.style.Theme_AppCompat_DayNight_Dialog);
        setContentView(R.layout.dialog_options_note);

        delete = findViewById(R.id.imageView4);
        password = findViewById(R.id.imageView3);
        favorite = findViewById(R.id.imageView2);
        favoriteColor = findViewById(R.id.textView4);
        imageView5 = findViewById(R.id.imageView5);
        lockText = findViewById(R.id.lock_text);

        final TextView lockText2 = findViewById(R.id.lock_text2);
        final TextView lockText3 = findViewById(R.id.lock_text3);
        final ImageView imageView5 = findViewById(R.id.imageView5);
        final ImageView imageView6 = findViewById(R.id.imageView6);
        final Space space = findViewById(R.id.space);
        final Space space2 = findViewById(R.id.space2);
        final Space space3 = findViewById(R.id.space3);
        final Space space4 = findViewById(R.id.space4);
        final Space space5 = findViewById(R.id.space5);
        final ImageView imageView = findViewById(R.id.imageView);

        final reSize2 size = new reSize2();
        final DisplayMetrics metrics = new DisplayMetrics();

        activity.getWindowManager().getDefaultDisplay().getMetrics(metrics);

        final int phoneWidth = metrics.widthPixels;
        final int phoneHeight = metrics.heightPixels;
        final int passwordWidth = (int) (phoneWidth/5d*3d);
        final int passwordHeight = (int) ((phoneHeight/50d*3d)*0.8d);
        final int spaceSize = Math.min((int) (passwordWidth/20d), (int) (passwordHeight/3d));

        size.reSize2(new Space[]{space, space2, space3, space4, space5}, spaceSize, spaceSize);
        size.reSize2(new View[]{password, delete, favorite}, passwordWidth, passwordHeight);
        size.reSize2(new View[]{lockText, lockText2, lockText3}, passwordWidth, (int) (passwordHeight*0.8d), true, true);
        size.reSize2(new View[]{favoriteColor, imageView6, imageView5}, (int) (spaceSize*1.5d), (int) (spaceSize*1.5d));

        size.reSize2(imageView, (int) (passwordWidth*1.05d), (int) (passwordHeight*3d + spaceSize*4));
    }

    public ImageView getDelete() {
        return delete;
    }

    public ImageView getFavorite() {
        return favorite;
    }

    public ImageView getPassword() {
        return password;
    }

    public ImageView getFavoriteColor() {
        return favoriteColor;
    }

    public ImageView getImageView5() {
        return imageView5;
    }

    public TextView getLockText() {
        return lockText;
    }

    public void build() {
        show();
    }
}

File: app/src/main/java/fr/newglace/notedesnazes/styles/dialog/OptionNote.kt
package fr.newglace.notedesnazes.styles.dialog

import android.app.Activity
import android.app.Dialog
import android.os.Build
import android.util.DisplayMetrics
import android.widget.ImageView
import android.widget.Space
import android.widget.TextView
import androidx.annotation.RequiresApi
import fr.newglace.notedesnazes.R
import fr.newglace.notedesnazes.styles.Colors
import fr.newglace.notedesnazes.styles.ReSize

class OptionNote @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP) constructor(activity: Activity) : Dialog(activity, R.style.Theme_AppCompat_DayNight_Dialog) {
    val delete: ImageView
    val password: ImageView
    val favorite: ImageView
    val favoriteColor: ImageView
    val imageView5: ImageView
    val lockText: TextView

    fun build() {
        show()
    }

    init {
        setContentView(R.layout.dialog_options_note)
        delete = findViewById(R.id.imageView4)
        password = findViewById(R.id.imageView3)
        favorite = findViewById(R.id.imageView2)
        favoriteColor = findViewById(R.id.textView4)
        imageView5 = findViewById(R.id.imageView5)
        lockText = findViewById(R.id.lock_text)
        val lockText2 = findViewById<TextView>(R.id.lock_text2)
        val lockText3 = findViewById<TextView>(R.id.lock_text3)
        val imageView5 = findViewById<ImageView>(R.id.imageView5)
        val imageView6 = findViewById<ImageView>(R.id.imageView6)
        val space = findViewById<Space>(R.id.space)
        val space2 = findViewById<Space>(R.id.space2)
        val space3 = findViewById<Space>(R.id.space3)
        val space4 = findViewById<Space>(R.id.space4)
        val space5 = findViewById<Space>(R.id.space5)
        val imageView = findViewById<ImageView>(R.id.imageView)
        val size = ReSize()
        val metrics = DisplayMetrics()
        activity.windowManager.defaultDisplay.getMetrics(metrics)
        val phoneWidth = metrics.widthPixels
        val phoneHeight = metrics.heightPixels
        val passwordWidth = (phoneWidth / 5.0 * 3.0).toInt()
        val passwordHeight = (phoneHeight / 50.0 * 3.0 * 0.8).toInt()
        val spaceSize = (passwordWidth / 20.0).toInt().coerceAtMost((passwordHeight / 3.0).toInt())
        size.reSizing(arrayOf(space, space2, space3, space4, space5), spaceSize, spaceSize)
        size.reSizing(arrayOf(password, delete, favorite), passwordWidth, passwordHeight)
        size.reSizing(arrayOf(lockText, lockText2, lockText3), passwordWidth, (passwordHeight * 0.8).toInt(), true, true)
        size.reSizing(arrayOf(favoriteColor, imageView6, imageView5), (spaceSize * 1.5).toInt(), (spaceSize * 1.5).toInt())
        size.reSizing(imageView, (passwordWidth * 1.05).toInt(), (passwordHeight * 3.0 + spaceSize * 4).toInt())
        val colors = Colors(activity)
        val draw = activity.getDrawable(R.drawable.bg_notes)
        val draw2 = activity.getDrawable(R.drawable.bg_notes)
        colors.editColor(imageView, 0, draw2)
        colors.editColor(delete, 1, draw)
        colors.editColor(password, 1, draw)
        colors.editColor(favorite, 1, draw)
    }
}